### DISCUSSION TITLE
Build Metadata process

### DISCUSSION THREAD

**Discussion by dvohra-igs** [usgs/gems-tools-pro#119] [Q&A] [answered]
I'm currently struggling with the Build Metadata tool in Arc Pro. The GeMS documentation that I'm seeing on building metadata talks about ArcCatalog, which is what I'm trying to fully move away from. If I've missed more current documentation somewhere, let me know! Currently, my workflow to produce metadata is:
1) Bring the database to level 3 with no errors aside from empty metadata.
2) Use Build Metadata to populate the structure of the metadata & some elements.
3) Use the USGS Metadata Wizard along with copying over appropriate elements from similar databases with complete metadata to finish producing the geodatabase-level metadata.
4) Using ArcCatalog, use the metadata xml produced from the previous step & feed it into the Step 2 tool along with the database to produce a xml file containing metadata for each individual feature dataset, feature class, and non-spatial table in the database.
5) Run a custom script that customizes the contents of the <purpose> and <abstract> tags to what my organization prefers.
6) Run Step 3 tool to embed each of the finalized metadata xml files into their correct position in the database.

What I'm having trouble with is using the Build Metadata tool in Arc Pro to do steps 4 & 6. Am I missing something, or is the intended workflow different from the steps above?

**Comment by ethoms-usgs** [comment, marked as answer]
To be honest, I didn't have any steps beyond your step 3 in mind when I wrote the tool. I was only thinking about getting a database-level metadata record out of it. For submission to the NGMDB, that's all that is required; metadata do not have be embedded and you don't have to produce a separate metadata file for every object in the gdb. If that is enough for your agency, you can call it done.

However, having since heard from many people that they want to embed the metadata as well as have the option to make files for child objects, I started a revision of the tool. With help from another programmer, I hope to post a new version of the toolbox with four new metadata tools within a couple months. My tool is not quite operational yet, but the others are and if you want to try the toolbox (or just the script files, if you want to call them in notebook, for example), I don't mind sharing it. In particular, there is a new Import Metadata tool that can parse the single database-level record that you end up with at the end of step 3 into a metadata record for each of the child objects and then embed the metadata.

Write to me at ethoms@usgs.gov if you want to do that.

**Reply by dvohra-igs** [reply]
> For submission to the NGMDB, that's all that is required; metadata do not have be embedded and you don't have to produce a separate metadata file for every object in the gdb. If that is enough for your agency, you can call it done.

Thanks for the clarification! For our own publication purposes, I think we're unlikely to move away from embedded metadata, but we'll see.
